pyang/translators/yang.py

```diff
@@ -231,19 +258,45 @@ def emit_stmt(ctx, stmt, fd, level, prev_kwd, prev_kwd_class, islast,
             if arg_on_new_line:
                 # arg was printed on a new line, increase indentation
                 n = 2
+
+            link_list['last'] = s
             emit_stmt(ctx, s, fd, level + 1, prev_kwd, kwd_class,
                       i == len(substmts),
-                      indent + (indentstep * n), indentstep)
-            kwd_class = get_kwd_class(s.keyword)
-            prev_kwd = s.keyword
-        fd.write(indent + '}\n')
+                      indent + (indentstep * n), indentstep, link_list)
+            if not is_line_end_comment(s):
+                kwd_class = get_kwd_class(s.keyword)
+                prev_kwd = s.keyword
+        fd.write(indent + '}')
+        last_substmt = link_list['last']
+        if last_substmt in link_list.keys():
+            last_substmt = link_list[last_substmt]
+        emit_line_end_comments_after(stmt, last_substmt, link_list, fd, True)
+        fd.write('\n')
 
     if (not islast and
         ((level == 1 and stmt.keyword in
           _keyword_with_trailing_blank_line_toplevel) or
          stmt.keyword in _keyword_with_trailing_blank_line)):
         fd.write('\n')
 
+def emit_line_end_comments_after(stmt, last_substmt, link_list, fd, need_same_level):
+    while last_substmt is not None:
+        is_sub_level = False
+        if not need_same_level:
+            is_sub_level = last_substmt.parent == stmt
+        if (is_line_end_comment(last_substmt) and
+                (last_substmt.parent == stmt.parent or (not need_same_level and is_sub_level))):
+            fd.write('  ' + last_substmt.arg)
```

Comment:
why two spaces here?  I think a line end comment should be printed as:

    } // one space between } and //

Reply:
OK，I will it with one space